src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,158 @@
+        }
+        System.out.println("________________________________________________________________________________________");
+    }
+
+    /**
+     * Prints duke chatbot message after user list tasks and list unfinished tasks at a date.
+     *
+     * @param command user command.
+     * @param str string containing a date.
+     * @param taskList task list to be printed.
+     */
+    public void printListCommandUi(String command, String str, TaskList taskList) {
+        // for list, list date, search
+        if (str.trim().equals("0")) { //list
+            System.out.println("________________________________________________________________________________"
+                    + "________");
+            System.out.println("Here are the tasks in your list:");
+            taskList.printTaskList();
+            System.out.println("________________________________________________________________________________"
+                    + "________");
+        } else if (str.matches("([0-9]{4})-([0-9]{2})-([0-9]{2})")) { //list date
+            System.out.println("________________________________________________________________________________"
+                    + "________");
+            System.out.println("Here are unfinished tasks on this date in your list:");
+            taskList.searchByDate(LocalDate.parse(str)).printTaskList();
+            System.out.println("________________________________________________________________________________"
+                    + "________");
+        } else {
+            System.out.println("________________________________________________________________________________"
+                    + "________");
+            System.out.println("Here are the matching tasks in your list:");
+            taskList.searchByKeyword(str).printTaskList();
+            System.out.println("________________________________________________________________________________"
+                    + "________");
+        }
+    }
+
+    /**
+     * Prints duke chatbot ending message that user sees when he/her stops/exits duke chatbot.
+     */
+    public void printEndingUi() {
+        System.out.println("________________________________________________________________________________________");
+        System.out.println("Chatbot stopped, all previous tasks will be auto-saved :D");
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println("________________________________________________________________________________________");
+    }
+
+    /**
+     * Prints message when task list storage loading occurs.
+     */
+    public void showLoadingError() {
+        System.out.println("________________________________________________________________________________________");
+        System.out.println("☹ OOPS!!! Loading error occurred, new text file is created for task list storage.");
```

Comment:
Perhaps some of these messages can be abstracted into a static constant as well to improve code readability?